File: specification/specs/x_pack/roll_up/create_rollup_job/CreateRollupJobRequest.ts
@rest_spec_name("rollup.put_job")
class CreateRollupJobRequest extends RequestBase {
  path_parts?: {
    id: Id;
  }
  query_parameters?: {
  }
  body?: {
    cron?: string;
    groups?: RollupGroupings;
    index_pattern?: string;
    metrics?: RollupFieldMetric[];
    page_size?: long;
    rollup_index?: IndexName;
  }
}

File: output/csharp/XPack/RollUp/CreateRollupJob/CreateRollupJobRequest.cs
using Nest.XPack;
using Nest.Internal;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class CreateRollupJobRequest : RequestBase {
		
		[DataMember(Name="cron")]
		public string Cron { get; set; }


		[DataMember(Name="groups")]
		public RollupGroupings Groups { get; set; }


		[DataMember(Name="index_pattern")]
		public string IndexPattern { get; set; }


		[DataMember(Name="metrics")]
		public List<RollupFieldMetric> Metrics { get; set; }


		[DataMember(Name="page_size")]
		public long PageSize { get; set; }


		[DataMember(Name="rollup_index")]
		public IndexName RollupIndex { get; set; }

	}
}
